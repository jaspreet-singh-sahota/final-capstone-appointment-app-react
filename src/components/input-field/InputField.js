import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles/inputField.module.css';

const InputField = ({
  type, name, placeholder, value, handlerOnChange,
}) => (
  <>
    <input
      className={styles.input}
      type={type}
      name={name}
      placeholder={placeholder}
      value={value}
      onChange={handlerOnChange}
      required
    />
  </>
);

InputField.propTypes = {
  type: PropTypes.string,
  name: PropTypes.string,
  placeholder: PropTypes.string,
  value: PropTypes.string,
  handlerOnChange: PropTypes.func,
};

InputField.defaultProps = {
  type: 'submit',
  name: '',
  placeholder: '',
  value: '',
  handlerOnChange: undefined,
};

export default InputField;
