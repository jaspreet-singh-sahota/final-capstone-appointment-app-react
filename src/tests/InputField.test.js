import React from 'react';
import { shallow } from 'enzyme';
import InputField from '../components/input-field/InputField';

describe('InputField', () => {
  it('should render a <input /> field', () => {
    const wrapper = shallow(<InputField />);
    expect(wrapper.find('input')).toHaveLength(1);
  });

  it('matches Render snapshot', () => {
    const wrapper = shallow(<InputField />);
    expect(wrapper).toMatchSnapshot();
  });
});
