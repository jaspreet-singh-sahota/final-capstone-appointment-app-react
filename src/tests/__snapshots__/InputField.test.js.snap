// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField matches Render snapshot 1`] = `ShallowWrapper {}`;
